src/components/Articles/SingleArticlePage.jsx

```diff
@@ -9,55 +9,82 @@ export default function SingleArticlePage() {
     const [article, setArticle] = useState({})
     const [comments, setComments] = useState([])
     const [isLoading, setIsLoading] = useState(false)
+    const [votes, setVotes] = useState(article.votes)
 
     useEffect(() => {
         setIsLoading(true)
         getArticle(article_id).then(article => {
             setArticle(article)
+            setVotes(article.votes)
+
         })
         getComments(article_id).then(comments => {
             setComments(comments)
             setIsLoading(false)
         })
     }, []);
 
+    const handleVotes = (vote) => {
+        patchVote(article_id, vote).then(article => {
+            setVotes(currVotes => currVotes + vote)
+        })
+    }
+
```

Comment:
This DOES work, but it's the non-optimistc approach. Ddeally we'd have our votes be [optimistically rendered](https://notes.northcoders.com/courses/js-front-end/ui-behaviours-and-optimistic-rendering), which would improve the user experience since there would be instant feedback.
This means we wouldn't wait for the response from patchVote, we'd increase the votes as soon as the vote button was pressed, assuming that the response was successful. We'd also have a .catch that would undo that in the case of setVotes failing.

Reply:
Gotcha! Thank you, I will get that implemented in the next task :)